src/flint/fmpz.jl

```diff
@@ -1003,7 +1003,7 @@ end
 """
 function root(x::fmpz, n::Int)
    x < 0 && iseven(n) && throw(DomainError("Argument $x must be positive if exponent $n is even"))
-   n <= 0 && throw(DomainError("Exponent must be non-negative: $n"))
+   n <= 0 && throw(DomainError("Exponent $n not positive in root"))
```

Comment:
When `DomainError` is used with one argument, it must be the "value" which is outside the valid domain. So I suggest: `DomainError(n, "Exponent not positive"))` (I prefer a bit "exponent must be positive" but it's not very important). Also, the stacktrace already indicates the function, so it's usually not necessary to also write it in the message, although it's not a problem.
For example, the Julia's `powmod` function has this message:
```
julia> powmod(10, -2, 2)
ERROR: DomainError with -2:
exponent must be >= 0
Stacktrace:
 [1] powmod(::Int64, ::Int64, ::Int64) at ...
[...]
```

Reply:
I don't know how to fix the previous DomainError one line up. By the way, what is the difference between an ArgumentError and a DomainError.

Reply:
It could be for example `DomainError((x, n), "Argument x must be positive if exponent is even"))`, but this would be fine I guess to keep interpolating `x` and `n` in the error string too.
The distinction between `DomainError` and `ArgumentError` is not so clear and I've seen people confused but it. I would say that `DomainError` is appropriate when the arguments are of the correct type, but their values is incompatible with what the function does, like here in this example. But honestly this is not used very consistently, even in `Base`. It seems that `DomainError` is more used for mathematical context.